src/closedai/pipelines.py

```diff
@@ -0,0 +1,66 @@
+# Helpful links:
+# https://huggingface.co/docs/transformers/main/en/main_classes/pipelines#transformers.TextGenerationPipeline
+
+
+import time
+from threading import Thread
+
+from transformers import TextIteratorStreamer, pipeline
+
+
+class ClosedAIPipeline:
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def generate_completion(self, text, **kwargs):
+        pass
+
+    def get_completion(self, text, **kwargs):
+        return "".join(list(self.generate_completion(text, **kwargs)))
+
+
+class DummyPipeline(ClosedAIPipeline):
+    def __init__(self, duration=5):
+        self.duration = duration
+
+    def generate_completion(self, text, **kwargs):
+        for i in range(1, self.duration + 1):
+            yield f", and {i}"
+            time.sleep(0.5)
+
+    def generate_chat_completion(self, messages, **kwargs):
+        for i in range(1, self.duration + 1):
+            yield f", and {i}"
+            time.sleep(0.5)
+
+
+class HuggingFacePipeline(ClosedAIPipeline):
```

Comment:
should be in separate file so we can avoid unnecessary imports with lazy loading. will do in future PR.